ui/src/app/utils/authentication.tsx

```diff
@@ -92,3 +91,37 @@ export function useAuthentication() {
 
   return {authLoaded, isSignedIn};
 }
+
+// The delay before continuing to avoid errors due to delays in applying the new scope grant
+const BILLING_SCOPE_DELAY_MS = 2000;
+
+const getAuthInstance = () => {
+  return gapi.auth2.getAuthInstance();
+};
+
+export const hasBillingScope = () => {
+  return getAuthInstance().currentUser.get().hasGrantedScopes('https://www.googleapis.com/auth/cloud-billing');
+};
+
+
+export const delay = ms => {
```

Comment:
duplicates https://github.com/all-of-us/workbench/blob/master/ui/src/app/utils/subscribable.ts#L3